providers/rpc/http.go

```diff
@@ -0,0 +1,77 @@
+package rpc
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// createRequest
+func (c *Config) createRequest(ctx context.Context, p RequestPayload) (*http.Request, error) {
+	var data []byte
+	if rj := c.Opts.Experimental.CustomRequestPayload; rj != nil && c.Opts.Experimental.DotPath != "" {
+		d, err := p.embedPayload(rj, c.Opts.Experimental.DotPath)
+		if err != nil {
+			return nil, err
+		}
+		data = d
+	} else {
+		d, err := json.Marshal(p)
+		if err != nil {
+			return nil, err
+		}
+		data = d
+	}
+
+	req, err := http.NewRequestWithContext(ctx, c.Opts.Request.HTTPMethod, c.listenerURL.String(), bytes.NewReader(data))
+	if err != nil {
+		return nil, err
+	}
+	for k, v := range c.Opts.Request.StaticHeaders {
+		req.Header.Add(k, strings.Join(v, ","))
+	}
+	if c.Opts.Request.HTTPContentType != "" {
+		req.Header.Set("Content-Type", c.Opts.Request.HTTPContentType)
+	}
+	if c.Opts.Request.TimestampHeader != "" {
+		req.Header.Add(c.Opts.Request.TimestampHeader, time.Now().Format(c.Opts.Request.TimestampFormat))
+	}
+
+	return req, nil
+}
+
+func (c *Config) handleResponse(resp *http.Response, reqKeysAndValues []interface{}) (ResponsePayload, error) {
+	kvs := reqKeysAndValues
+	defer func() {
+		if !c.LogNotificationsDisabled {
+			kvs = append(kvs, responseKVS(resp)...)
+			c.Logger.Info("rpc notification details", kvs...)
+		}
+	}()
+	defer resp.Body.Close()
+	bodyBytes, err := io.ReadAll(resp.Body)
```

Comment:
Should there be limit/check for the response body size, just to make sure the provider isn't vulnerable to being dos'ed ?

Reply:
sure. Any suggestions on how to define an upper limit? The `ResponsePayload.Result` is of type `interface{}` so to allow for different methods to return the types they need. This field could be of variable length. And then we have some string fields that could be large or small depending on the consumer.

Reply:
Hey @joelrebel , i've added `maxContentLenAllowed` and a content length check. The response contract we've defined is pretty small and my tests showed that currently only about 50 bytes are needed. With that in mind i made `maxContentLenAllowed` equal to 512KB. Thoughts?